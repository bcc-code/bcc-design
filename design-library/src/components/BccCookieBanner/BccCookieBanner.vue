<script setup lang="ts">
import { CookieIcon } from "@bcc-code/icons-vue";
import BccButton from "../BccButton/BccButton.vue";

type Props = {
  open: boolean;
  title?: string;
  showOverlay?: boolean;
  declineButtonText?: string;
  acceptButtonText?: string;
};

withDefaults(defineProps<Props>(), {
  open: false,
  showOverlay: true,
  declineButtonText: "Decline",
  acceptButtonText: "Accept",
});

const emit = defineEmits(["accept", "decline"]);
</script>

<template>
  <Transition name="fade" v-if="showOverlay">
    <div v-if="open" class="bcc-cookie-banner-overlay"></div>
  </Transition>
  <Transition name="fade-fast">
    <div v-if="open" class="bcc-cookie-banner-wrapper">
      <div class="bcc-cookie-banner">
        <div class="bcc-cookie-banner-content">
          <div class="bcc-cookie-banner-title">
            <CookieIcon class="bcc-cookie-banner-icon" />
            <h4>{{ title }}</h4>
          </div>

          <p>
            <slot />
          </p>
        </div>

        <div class="bcc-cookie-banner-actions">
          <BccButton
            variant="secondary"
            @click="emit('decline')"
            class="bcc-cookie-banner-decline-button"
          >
            {{ declineButtonText }}
          </BccButton>
          <BccButton @click="emit('accept')" class="bcc-cookie-banner-accept-button">
            {{ acceptButtonText }}
          </BccButton>
        </div>
      </div>
    </div>
  </Transition>
</template>
