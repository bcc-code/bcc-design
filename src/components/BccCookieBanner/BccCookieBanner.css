@layer components {
    .bcc-cookie-banner {
        @apply mx-auto text-body-base flex flex-col justify-between gap-6 rounded-lg bg-white p-4 text-primary shadow-lg md:gap-8 md:flex-row max-w-7xl;
    }

    /* Overlay/Wrapper */
    .bcc-cookie-banner-wrapper {
        @apply fixed inset-x-0 bottom-0 z-50 p-4;
    }

    .bcc-cookie-banner-overlay {
        @apply fixed inset-0 z-50 bg-black/80 backdrop-blur-sm transition-opacity;
    }

    /* Content */
    .bcc-cookie-banner-content {
        @apply flex flex-col gap-2;
    }

    .bcc-cookie-banner-icon {
        @apply h-6 w-6;
    }

    .bcc-cookie-banner-title {
        @apply flex items-center gap-2 text-heading-medium;
    }

    .bcc-cookie-banner a {
        @apply underline text-interactive hover:text-interactive-hover active:text-interactive-active;
    }

    /* Actions */
    .bcc-cookie-banner-actions {
        @apply flex flex-col justify-center gap-4 sm:flex-row md:items-end;
    }
    .bcc-cookie-banner-decline-button {
        @apply order-3 flex-1 sm:order-1;
    }
    .bcc-cookie-banner-accept-button {
        @apply order-2 flex-1;
    }
}
