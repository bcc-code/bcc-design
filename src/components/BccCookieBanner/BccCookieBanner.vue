<script setup lang="ts">
import { CookieIcon } from "@bcc-code/icons-vue";
import BccButton from "../BccButton/BccButton.vue";

type Props = {
  open: boolean;
  showOverlay?: boolean;
  declineButtonText?: string;
  acceptButtonText?: string;
};

withDefaults(defineProps<Props>(), {
  open: false,
  showOverlay: true,
  declineButtonText: "Decline",
  acceptButtonText: "Accept",
});

const emit = defineEmits(["accept", "decline"]);
</script>

<template>
  <Transition name="fade" v-if="showOverlay">
    <div v-if="open" class="bcc-cookie-banner-overlay"></div>
  </Transition>
  <Transition name="fade-fast">
    <div v-if="open" :class="{ 'bcc-cookie-banner-wrapper': showOverlay }">
      <div class="bcc-cookie-banner">
        <CookieIcon class="bcc-cookie-banner-icon" />
        <p>
          <slot />
        </p>

        <div class="bcc-cookie-banner-actions">
          <BccButton variant="secondary" @click="emit('decline')">
            {{ declineButtonText }}
          </BccButton>
          <BccButton @click="emit('accept')">{{ acceptButtonText }}</BccButton>
        </div>
      </div>
    </div>
  </Transition>
</template>
