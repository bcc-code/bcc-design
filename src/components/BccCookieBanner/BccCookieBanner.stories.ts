import BccCookieBanner from "./BccCookieBanner.vue";
import BccButton from "../BccButton/BccButton.vue";

import type { Meta, StoryFn } from "@storybook/vue3";

/**
 * A cookie banner with an accept and decline button. Emits `decline` and `accept` events when the buttons are clicked.
 */
export default {
  title: "Components/BccCookieBanner",
  component: BccCookieBanner,
  argTypes: {
    showOverlay: {
      description: "Controls whether to overlay a dark background behind the cookie banner",
    },
  },
} as Meta<typeof BccCookieBanner>;

const Template: StoryFn<typeof BccCookieBanner> = (args) => ({
  components: { BccCookieBanner, BccButton },
  setup() {
    return { args };
  },
  template: `
    <BccCookieBanner v-bind="args" @decline="args.open = false" @accept="args.open = false">
      We would like to evaluate your usage of this app in order to improve your experience. Read
      more in our <a href="https://event.bcc.no/personvernerklaering/" target="_blank">privacy policy</a>.
    </BccCookieBanner>
    
    <BccButton variant="secondary" @click="args.open=true">Open cookie banner</BccButton>
  `,
});

export const Example = Template.bind({});
Example.args = {
  open: false,
  title: "We value your privacy",
  showOverlay: true,
  declineButtonText: "Decline",
  acceptButtonText: "Accept",
};
Example.parameters = {
  docs: {
    source: {
      language: "html",
      code: `
<BccCookieBanner title="We value your privacy" :open="showCookieBanner" @decline="showCookieBanner = false" @accept="showCookieBanner = false">
  We would like to evaluate your usage of this app in order to improve your experience.
  Read more in our <a href="https://event.bcc.no/personvernerklaering/" target="_blank">privacy policy</a>.
</BccCookieBanner>
    `,
    },
  },
};
