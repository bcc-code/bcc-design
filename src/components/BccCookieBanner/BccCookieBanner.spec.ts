import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import BccCookieBanner from "./BccCookieBanner.vue";

describe("BccCookieBanner", () => {
  it("renders with an overlay", async () => {
    const wrapper = mount(BccCookieBanner, { props: { open: true, showOverlay: true } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders without an overlay", async () => {
    const wrapper = mount(BccCookieBanner, { props: { open: true, showOverlay: false } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders slot and button text", async () => {
    const wrapper = mount(BccCookieBanner, {
      props: { open: true, acceptButtonText: "I Accept" },
      slots: { default: "Default slot content" },
    });
    expect(wrapper.text()).toContain("I Accept");
    expect(wrapper.text()).toContain("Default slot content");
  });
});
