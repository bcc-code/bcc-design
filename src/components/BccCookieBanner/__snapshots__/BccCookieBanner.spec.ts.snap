// Vitest Snapshot v1

exports[`BccCookieBanner > renders with an overlay 1`] = `
"<transition-stub name=\\"fade\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
  <div class=\\"bcc-cookie-banner-overlay\\"></div>
</transition-stub>
<transition-stub name=\\"fade-fast\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
  <div class=\\"bcc-cookie-banner-wrapper\\">
    <div class=\\"bcc-cookie-banner\\">
      <div class=\\"bcc-cookie-banner-content\\">
        <div class=\\"bcc-cookie-banner-title\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" aria-hidden=\\"true\\" viewBox=\\"0 96 960 960\\" fill=\\"currentColor\\" class=\\"bcc-cookie-banner-icon\\">
            <path d=\\"M420.118 498Q446 498 464 479.882q18-18.117 18-44Q482 410 463.882 392q-18.117-18-44-18Q394 374 376 392.118q-18 18.117-18 44Q358 462 376.118 480q18.117 18 44 18Zm-80 200Q366 698 384 679.882q18-18.117 18-44Q402 610 383.882 592q-18.117-18-44-18Q314 574 296 592.118q-18 18.117-18 44Q278 662 296.118 680q18.117 18 44 18ZM600 736q17 0 28.5-11.5T640 696q0-17-11.5-28.5T600 656q-17 0-28.5 11.5T560 696q0 17 11.5 28.5T600 736ZM480.162 980.065q-83.119 0-156.836-31.858-73.717-31.859-128.696-86.837-54.978-54.979-86.837-128.741Q75.935 658.867 75.935 576q0-84.63 32.337-158.348 32.337-73.717 87.174-128.598 54.837-54.88 127.315-86.641 72.478-31.761 153-32.478 33.913 0 54.967 15.956 21.055 15.957 25.816 49.631 7.76 57.282 47.967 96.467 40.206 39.185 96.728 47.663 21.957 3 34.913 16.837 12.957 13.837 18.435 39.033 5.717 29.326 25.402 52.869 19.685 23.544 46.772 37.065 33.416 16.136 46.36 38.242 12.944 22.106 8.944 56.737-7.717 74.545-41.956 139.99-34.239 65.445-87.979 114.303-53.739 48.859-123.184 77.098-69.445 28.239-148.784 28.239Zm-.091-68.13q139.814 0 232.122-91.587 92.307-91.587 104.263-225.957-53.043-20-86.304-59.261-33.261-39.26-41.261-89.543-80.043-11.239-135.902-69.761-55.859-58.522-64.576-136.282-70.891-2.044-133.717 24-62.826 26.043-109.609 71.728-46.783 45.685-73.902 108.072-27.12 62.387-27.12 132.656 0 140.431 97.752 238.183 97.752 97.752 238.254 97.752Zm.69-340.218Z\\"></path>
          </svg>
          <h4></h4>
        </div>
        <p></p>
      </div>
      <div class=\\"bcc-cookie-banner-actions\\"><button class=\\"bcc-button bcc-button-secondary bcc-button-center bcc-cookie-banner-decline-button\\">
          <!--v-if--><span class=\\"order-3\\">Decline</span>
        </button><button class=\\"bcc-button bcc-button-primary bcc-button-center bcc-cookie-banner-accept-button\\">
          <!--v-if--><span class=\\"order-3\\">Accept</span>
        </button></div>
    </div>
  </div>
</transition-stub>"
`;

exports[`BccCookieBanner > renders without an overlay 1`] = `
"<!--v-if-->
<transition-stub name=\\"fade-fast\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
  <div class=\\"bcc-cookie-banner-wrapper\\">
    <div class=\\"bcc-cookie-banner\\">
      <div class=\\"bcc-cookie-banner-content\\">
        <div class=\\"bcc-cookie-banner-title\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" aria-hidden=\\"true\\" viewBox=\\"0 96 960 960\\" fill=\\"currentColor\\" class=\\"bcc-cookie-banner-icon\\">
            <path d=\\"M420.118 498Q446 498 464 479.882q18-18.117 18-44Q482 410 463.882 392q-18.117-18-44-18Q394 374 376 392.118q-18 18.117-18 44Q358 462 376.118 480q18.117 18 44 18Zm-80 200Q366 698 384 679.882q18-18.117 18-44Q402 610 383.882 592q-18.117-18-44-18Q314 574 296 592.118q-18 18.117-18 44Q278 662 296.118 680q18.117 18 44 18ZM600 736q17 0 28.5-11.5T640 696q0-17-11.5-28.5T600 656q-17 0-28.5 11.5T560 696q0 17 11.5 28.5T600 736ZM480.162 980.065q-83.119 0-156.836-31.858-73.717-31.859-128.696-86.837-54.978-54.979-86.837-128.741Q75.935 658.867 75.935 576q0-84.63 32.337-158.348 32.337-73.717 87.174-128.598 54.837-54.88 127.315-86.641 72.478-31.761 153-32.478 33.913 0 54.967 15.956 21.055 15.957 25.816 49.631 7.76 57.282 47.967 96.467 40.206 39.185 96.728 47.663 21.957 3 34.913 16.837 12.957 13.837 18.435 39.033 5.717 29.326 25.402 52.869 19.685 23.544 46.772 37.065 33.416 16.136 46.36 38.242 12.944 22.106 8.944 56.737-7.717 74.545-41.956 139.99-34.239 65.445-87.979 114.303-53.739 48.859-123.184 77.098-69.445 28.239-148.784 28.239Zm-.091-68.13q139.814 0 232.122-91.587 92.307-91.587 104.263-225.957-53.043-20-86.304-59.261-33.261-39.26-41.261-89.543-80.043-11.239-135.902-69.761-55.859-58.522-64.576-136.282-70.891-2.044-133.717 24-62.826 26.043-109.609 71.728-46.783 45.685-73.902 108.072-27.12 62.387-27.12 132.656 0 140.431 97.752 238.183 97.752 97.752 238.254 97.752Zm.69-340.218Z\\"></path>
          </svg>
          <h4></h4>
        </div>
        <p></p>
      </div>
      <div class=\\"bcc-cookie-banner-actions\\"><button class=\\"bcc-button bcc-button-secondary bcc-button-center bcc-cookie-banner-decline-button\\">
          <!--v-if--><span class=\\"order-3\\">Decline</span>
        </button><button class=\\"bcc-button bcc-button-primary bcc-button-center bcc-cookie-banner-accept-button\\">
          <!--v-if--><span class=\\"order-3\\">Accept</span>
        </button></div>
    </div>
  </div>
</transition-stub>"
`;
